src/main/java/webApp/webController.java

```diff
@@ -1,56 +1,62 @@
 package webApp;
 
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.context.annotation.ComponentScan;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.*;
-import repository.SurveyRepository;
+import survey.OpenEndedQuestion;
 import survey.Survey;
 import survey.Question;
 
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Iterator;
-import java.util.List;
 
 @Controller
 public class webController {
     @Autowired
     private SurveyRepository repo;
 
-    @GetMapping("/index")
+    @GetMapping("/")
     public String index(Model model){
-        return "index";
+       return "root";
     }
 
-    @PostMapping("/createSurvey")
-    public String createSurvey(@RequestParam("surveyName") String surveyName, @RequestBody Collection<Question> questions) {
-        Survey survey = new Survey(surveyName);
-        survey.setQuestions(questions);
+    @PostMapping(value = "/createSurvey", consumes = "application/json", produces = "application/json")
+    @ResponseBody
+    public Response createSurvey(@RequestBody SurveyMessage surveyMessage) {
+        Survey survey = new Survey(surveyMessage.getName());
+        Collection<Question> questionList = new ArrayList<Question>();
+
+        for (QuestionMessage question : surveyMessage.getQuestions()) {
```

Comment:
I am wondering if we compress this by using lambdas and filters (java 8)

Reply:
questionList.add(surveyMessage.getQuestions().stream().filter(question -> question.getType().equals("openEnded")).collect(Collectors.toList()));